src/AoC2024/1/aoc24_1.cpp

```diff
@@ -1,7 +1,66 @@
-#include <iostream>
+#include <cstdlib>
+#include <fstream>
+#include <sstream>
+#include <string>
 
-int main(){
-    std::cout << "Hello world!" << '\n';
+#include "aoc24_1.h"
 
-    return 0;
+std::vector<int>
+calculateDistances(const std::vector<std::pair<int,int>>& preparedList) {
+    std::vector<int> distances = {};
+
+    const bool listIsNotEmpty = preparedList.size() > 0;
+    if(listIsNotEmpty)
+        std::for_each(preparedList.begin(), preparedList.end(), [&](const std::pair<int,int>& pair){
+            int left = std::get<0>(pair);
+            int right = std::get<1>(pair);
+            int distance = std::abs(left - right);
+            distances.push_back(distance);
+        });
+    
+    return distances;
```

Comment:
Am I right, if I understand this as:
Since C++17 copy elision is likely to occur even for named objects
But `std::move(distances)` would still be a best practice, since it forces the move instead of a copy?

I'm not exactly sure why `std::move` is still recommended in recent C++ versions (since 17), the page reads like this is already a pretty strong "promise".

Reply:
`std::` is an indication to the compiler. copy elision is part of all of this. you can basically use std::move for complex types and not use it for basic types. most of the time the measurable performance impact should be negligible.